x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/job/process/autodetect/output/AutodetectResultProcessor.java

```diff
@@ -87,7 +90,9 @@
     private final FlushListener flushListener;
     private volatile boolean processKilled;
     private volatile boolean failed;
-    private int bucketCount; // only used from the process() thread, so doesn't need to be volatile
+    private long priorRunsBucketCount;
+    private int currentRunBucketCount; // only used from the process() thread, so doesn't need to be volatile
```

Comment:
should we take this chance and make this a `long` as well? It seems like something that could hit overflow problems.

Reply:
With a 1 second bucket span it will take 68 years to overflow, so pretty unlikely.  But maybe I can avoid some casting by making it `long`, which will make the code cleaner.  If so I'll change it.

Reply:
There wasn't much casting but I changed it anyway just so both variables have the same type.